iceoryx_posh/test/moduletests/test_posh_runtime.cpp

```diff
@@ -545,6 +546,276 @@ TEST_F(PoshRuntime_test, GetMiddlewareSubscriberWithQueueFullPolicySetToBlockPub
+}
+
+TEST_F(PoshRuntime_test, GetMiddlewareClientWithQueueGreaterMaxCapacityClampsQueueToMaximum)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "8e34f962-e7c9-40ac-9796-a12f92c4d674");
+    constexpr uint64_t MAX_QUEUE_CAPACITY = iox::popo::ClientChunkQueueConfig::MAX_QUEUE_CAPACITY;
+    const iox::capro::ServiceDescription sd{"919", "1", "20"};
+    iox::popo::ClientOptions clientOptions;
+    clientOptions.responseQueueCapacity = MAX_QUEUE_CAPACITY + 1U;
+
+    auto clientPort = m_runtime->getMiddlewareClient(sd, clientOptions);
+
+    ASSERT_THAT(clientPort, Ne(nullptr));
+    EXPECT_EQ(clientPort->m_chunkReceiverData.m_queue.capacity(), MAX_QUEUE_CAPACITY);
+}
+
+TEST_F(PoshRuntime_test, GetMiddlewareClientWithQueueCapacityZeroClampsQueueCapacityTo1)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "7b6ffd68-46d4-4339-a0df-6fecb621f765");
+    const iox::capro::ServiceDescription sd{"99", "19", "20"};
+    iox::popo::ClientOptions clientOptions;
+    clientOptions.responseQueueCapacity = 0U;
+
+    auto clientPort = m_runtime->getMiddlewareClient(sd, clientOptions);
+
+    ASSERT_THAT(clientPort, Ne(nullptr));
+    EXPECT_EQ(clientPort->m_chunkReceiverData.m_queue.capacity(), 1U);
+}
+
+TEST_F(PoshRuntime_test, GetMiddlewareClientWhenMaxClientsAreUsedResultsInClientlistOverflow)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "6f2de2bf-5e7e-47b1-be42-92cf3fa71ba6");
+    auto clientOverflowDetected{false};
+    auto errorHandlerGuard = iox::ErrorHandler::setTemporaryErrorHandler(
+        [&](const iox::Error error, const std::function<void()>, const iox::ErrorLevel) {
+            if (error == iox::Error::kPORT_POOL__CLIENTLIST_OVERFLOW)
+            {
+                clientOverflowDetected = true;
+            }
+        });
+
+    uint32_t i{0U};
+    for (; i < iox::MAX_CLIENTS; ++i)
+    {
+        auto clientPort = m_runtime->getMiddlewareClient(
+            iox::capro::ServiceDescription(iox::capro::IdString_t(TruncateToCapacity, convert::toString(i)),
+                                           iox::capro::IdString_t(TruncateToCapacity, convert::toString(i + 1U)),
+                                           iox::capro::IdString_t(TruncateToCapacity, convert::toString(i + 2U))));
+        ASSERT_THAT(clientPort, Ne(nullptr));
+    }
+    EXPECT_FALSE(clientOverflowDetected);
+
+    auto clientPort = m_runtime->getMiddlewareClient(
+        iox::capro::ServiceDescription(iox::capro::IdString_t(TruncateToCapacity, convert::toString(i)),
+                                       iox::capro::IdString_t(TruncateToCapacity, convert::toString(i + 1U)),
+                                       iox::capro::IdString_t(TruncateToCapacity, convert::toString(i + 2U))));
+    EXPECT_THAT(clientPort, Eq(nullptr));
+    EXPECT_TRUE(clientOverflowDetected);
+}
+
+TEST_F(PoshRuntime_test, GetMiddlewareClientWithInvalidNameLeadsToTermination)
```

Comment:
```suggestion
TEST_F(PoshRuntime_test, GetMiddlewareClientWithInvalidNameLeadsToError)
```

Reply:
Oh I was wondering what kind of name you meant therefore here comes the next suggestion:
`GetMiddlewareClientWithInvalidNodeNameLeadsToTermination` <- `NodeName`